src/app/reporting/ReportSchedulerImpl.cpp

```diff
@@ -68,11 +68,6 @@ void ReportSchedulerImpl::OnReadHandlerCreated(ReadHandler * aReadHandler)
                     "Registered a ReadHandler that will schedule a report between system Timestamp: %" PRIu64
                     " and system Timestamp %" PRIu64 ".",
                     newNode->GetMinTimestamp().count(), newNode->GetMaxTimestamp().count());
-
-    Milliseconds32 newTimeout;
```

Comment:
Hold on.  OnReadHandlerSubscribed we _do_ want to recompute the timeouts, no?   

Also, if we call this after SendSubscribeResponse (as we do per the changes above), that means the OnSubscriptionAction call in ReadHandler::SendSubscribeResponse does not really work, right?  I'm _really_ surprised (and dismayed) this is passing tests.

It seems like the set of actual notifications we want here is:

1. OnReadHandlerSubscribed (which can happen in SendSubscribeResponse, or where you put it, as long as it's after we remove the priming flag).
2. OnSubscriptionReportFinished (which is the current OnSubscriptionAction call in ReadHandler::SendReportData).

Reply:
Initially I put it back, until I realized there is a redundancy here, after we call `mObserver->OnReadHandlerSubscribed(this);` we call `MoveToState(HandlerState::CanStartReporting);`, which will call `mObserver->OnBecameReportable(this);` if we are in a subscription state.

mObserver->OnBecameReportable(this); will schedule the next report.

Reply:
Ah, I see.  That's worth documenting in OnReadHandlerSubscribed.

And we should still take out the OnSubscriptionAction in SendSubscribeResponse, sounds like.

Reply:
Indeed, taken out and renamed to OnSubscriptionReport, comment added.